src/font/lv_font_cache.c

```diff
@@ -0,0 +1,212 @@
+    uint32_t glyph_index;
+
+    /* value */
+    lv_draw_buf_t * draw_buf;
+} font_bitmap_cache_data_t;
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static inline bool font_is_built_in(const lv_font_t * font);
+static void * get_bitmap_cached(lv_font_glyph_dsc_t * g_dsc, const lv_font_fmt_txt_dsc_t * fdsc, uint32_t gid);
+static bool font_bitmap_create_cb(font_bitmap_cache_data_t * data, void * user_data);
+static void font_bitmap_free_cb(font_bitmap_cache_data_t * data, void * user_data);
+static lv_cache_compare_res_t font_bitmap_compare_cb(const font_bitmap_cache_data_t * lhs,
+                                                     const font_bitmap_cache_data_t * rhs);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_font_cache_init(uint32_t max_glyph_cnt)
+{
+    LV_ASSERT_MSG(font_bitmap_cache == NULL, "Font cache already initialized");
+
+    lv_cache_ops_t ops = {
+        .compare_cb = (lv_cache_compare_cb_t)font_bitmap_compare_cb,
+        .create_cb = (lv_cache_create_cb_t)font_bitmap_create_cb,
+        .free_cb = (lv_cache_free_cb_t)font_bitmap_free_cb,
+    };
+
+    lv_cache_t * cache = lv_cache_create(&lv_cache_class_lru_rb_count, sizeof(font_bitmap_cache_data_t),
+                                         max_glyph_cnt, ops);
+    lv_cache_set_name(cache, "FONT_BITMAP");
+
+    font_bitmap_cache = cache;
+}
+
+void lv_font_cache_deinit(void)
+{
+    lv_cache_destroy(font_bitmap_cache, NULL);
+    font_bitmap_cache = NULL;
+}
+
+const void * lv_font_cache_get_glyph_bitmap(lv_font_glyph_dsc_t * g_dsc, lv_draw_buf_t * draw_buf)
+{
+    LV_ASSERT_NULL(g_dsc);
+
+    const lv_font_t * font = g_dsc->resolved_font;
+    if(!font_is_built_in(font)) {
+        /* If the font is not built-in, use the original function to get the bitmap */
+        return font->get_glyph_bitmap(g_dsc, draw_buf);
+    }
+
+    lv_font_fmt_txt_dsc_t * fdsc = (lv_font_fmt_txt_dsc_t *)font->dsc;
+    uint32_t gid = g_dsc->gid.index;
+    if(!gid) return NULL;
+
+    const lv_font_fmt_txt_glyph_dsc_t * gdsc = &fdsc->glyph_dsc[gid];
+    if(g_dsc->req_raw_bitmap) return &fdsc->glyph_bitmap[gdsc->bitmap_index];
+
+    int32_t gsize = (int32_t) gdsc->box_w * gdsc->box_h;
+    if(gsize == 0) return NULL;
+
+    return get_bitmap_cached(g_dsc, fdsc, gid);
+}
+
+void lv_font_cache_release_glyph(const lv_font_t * font, lv_font_glyph_dsc_t * g_dsc)
+{
+    LV_ASSERT_NULL(font);
+    LV_ASSERT_NULL(g_dsc);
+
+    if(!font_is_built_in(font)) {
+        /*Font cache can be released only for built-in fonts*/
+        return;
+    }
+
+    lv_cache_release(font_bitmap_cache, g_dsc->entry, NULL);
+    g_dsc->entry = NULL;
+}
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+static inline bool font_is_built_in(const lv_font_t * font)
```

Comment:
I suggest not treating the built-in fonts in a special way. It seems here the gist is if the font is static or not.

Reply:
At this stage, the built-in static font has not added this static_bitmap tag. I first use this function to judge. Although it is a bit strange, it can correctly distinguish and is forward compatible. The body of this function explains the situation.

Reply:
I see. Then let's name this function `has_static_bitmap`.